### PR TITLE
Group all pseudo nodes from a given host, talking to the same node, together

### PR DESCRIPTION
**Pull request by tomwilkie**

### CODE REVIEW

**Comment by tomwilkie** [comment]
Fixes #99

File: report/topology_test.go [hunk 1]
```diff
@@ -14,13 +14,17 @@ func init() {
...
+	unknownClient3 = ScopeDelim + "10.10.10.11" + ScopeDelim + "54020" // Check this one isn't deduped
+	server80       = ScopeDelim + "192.168.1.1" + ScopeDelim + "80"    // apache
 
-	clientIP = ScopeDelim + "10.10.10.20"
-	serverIP = ScopeDelim + "192.168.1.1"
-	randomIP = ScopeDelim + "172.16.11.9" // only in Network topology
+	clientIP    = ScopeDelim + "10.10.10.20"
+	serverIP    = ScopeDelim + "192.168.1.1"
+	randomIP    = ScopeDelim + "172.16.11.9" // only in Network topology
+	unknownHost = ScopeDelim + "10.10.10.10"
```
Comment (peterbourgon) [inline report/topology_test.go]: `unknownIP`

File: report/topology_test.go [hunk 2]
```diff
@@ -140,19 +164,42 @@ func TestRenderByProcessPID(t *testing.T) {
...
-			Rank:        "215",
-			Pseudo:      false,
-			Adjacency:   NewIDList("pid:client-54001-domain:10001", "pid:client-54002-domain:10001"),
+			ID:         "pid:server-80-domain:215",
+			LabelMajor: "apache",
+			LabelMinor: "server-80-domain (215)",
+			Rank:       "215",
+			Pseudo:     false,
+			Adjacency: NewIDList("pid:client-54001-domain:10001", "pid:client-54002-domain:10001",
+				"pseudo:;10.10.10.10;192.168.1.1;80", "pseudo:;10.10.10.11;192.168.1.1;80"),
```
Comment (peterbourgon) [inline report/topology_test.go]: Please either don't break or break on each parameter, e.g.

```go
Adjacency: NewIDList(
    "pid:client-xxx",
    "pid:client-yyy",
    "pid:client-zzz",
),
```

**Comment by peterbourgon** [comment]
Nice expansion of the test coverage. :+1:

**Pull request merged**